apps/www/_blog/2023-04-14-pg-tle.mdx

```diff
@@ -0,0 +1,66 @@
+
+## Trusted Language Extensions
+
+Near the end of 2022, we got a surprise gift from the team at Amazon Web Services, a new Postgres extension called `pg_tle`, that allows you to **Trusted Language Extensions (TLEs).** A TLE is just like a regular Postgres extension, except that **does not need to be installed on the server filesystem** it is written in a **trusted language** you can install extensions from any database client using the function `pgtle.install_extension()`, just like you can create other objects like tables and views.
+
+Of course, TLEs _must_ be written in trusted languages as described above, hence the name, so no extensions that are written in C can be TLEs. There are still many extensions that are already written in trusted languages, and soon many more to come with some exciting progress in more performant trusted languages like [pl/rust](https://github.com/tcdi/plrust) which recently reached an official 1.0.0 version release. By removing the constraint that extensions must be installed on the server filesystem, TLEs open up a whole new world of possibilities for packaging and distributing new and modular functionality for Postgres.
+
+pg_tle is available on new Supabase projects and on Amazon RDS for Postgres 14.5 and higher. Supabase also includes an extension called [pgsql-http](https://github.com/pramsey/pgsql-http) for making HTTP calls from SQL to REST APIs. So, naturally putting two and two together, we thought it would be cool to see if it’s possible to install TLEs from a REST API using only SQL!
+
+You can, and we’ll show you how downloading TLEs directly into your database turns out not only to be possible but also very powerful! In this blog post we will present a prototype for this idea, and our future plans to extend TLEs into a full _community ecosystem_ of installable packages, just like many other languages enjoy. pip for Python, npm for Javascript, cargo for Rust, cpan for Perl, and now dbdev for Postgres!
```

Comment:
I think these two paragraphs should go below "The Extension Shortfall" - because it leads directly onto the dbdev paragraphs